packages/engine/src/worker/runner/python/hash_util.py

```diff
@@ -1,20 +1,19 @@
 from json import loads
+
 import pyarrow as pa
 from pyarrow.types import is_primitive
-from wrappers import np_force_writable
-
 
-def json_deepcopy(x):
-    return deepcopy(x)
+from wrappers import np_force_writable
 
 
-def load_shallow(vector):
-    return [elem for elem in vector]
+# TODO: Change arguments after upgrading Arrow.
+def load_shallow(vector, _is_nullable, _is_any):
```

Comment:
Do we need this function at all? It seems like it's just there for consistency with the JS impl and I'm not sure why

Reply:
This function is used in several places (also as function parameters). I also tried removing it, but it wasn't straightforward and didn't improve readability. I'd just leave it, especially if we want to use the other parameters later on.

Reply:
Removing `load_shallow` from `loaders` will default to just return the element instead of calling a loader, so it's fine to remove it (I removed it).